src/Microsoft.AspNetCore.Sockets.Client/Connection.cs

```diff
@@ -59,7 +65,13 @@ public Task StartAsync(ITransport transport, HttpClient httpClient)
                     new InvalidOperationException("Cannot start a connection that is not in the Initial state."));
             }
 
-            StartAsyncInternal(transport, httpClient)
+            if (httpClient == null)
+            {
+                // We are creating the client so store it to be able to dispose
+                _httpClient = httpClient = new HttpClient();
+            }
+
+            StartAsyncInternal(transportFactory ?? new DefaultTransportFactory(TransportType.All, _loggerFactory), httpClient)
```

Comment:
tcs makes it easy to avoid race - https://github.com/aspnet/SignalR/commit/23375c911b4659d171bf8532b824c32771a54cb1 contains full details